### PR TITLE
Solana engine

### PR DESCRIPTION
**Pull request by swimalick**
Tested by running `propellerInitiate` commands from BSC -> Solana

### CODE REVIEW

File: relayer/spy_relayer/src/relayer/solana.ts [hunk 1]
```diff
@@ -44,6 +58,28 @@ export async function relaySolana(
...
+    solanaProvider,
+  );
+
+  const twoPoolProgram = new Program(
+    idl.twoPool,
+    env.swimTwoPoolAddress,
+    solanaProvider
+  );
+
+  const propellerEngineAnchorProvider = new AnchorProvider(
```
Comment (wormat) [inline relayer/spy_relayer/src/relayer/solana.ts]: Why do you need a second provider?

File: relayer/spy_relayer/src/relayer/solana.ts [hunk 2]
```diff
@@ -150,22 +188,90 @@ export async function relaySolana(
...
+
+    logger.debug("starting processSwimPayloadTxn");
+    const processSwimPayloadTxn = swimTxns[swimTxnIndex++];
+    const processSwimPayloadTxnSig =
+      await propellerEngineAnchorProvider.sendAndConfirm(
+        processSwimPayloadTxn,
+        [keypair],
+      );
+    logger.debug(`processSwimPayloadTxn complete ${processSwimPayloadTxnSig}`);
+  }
```
Comment (wormat) [inline relayer/spy_relayer/src/relayer/solana.ts]: Should be simple to make this DRYer

File: relayer/spy_relayer/src/relayer/solana_utils.ts [hunk 3]
```diff
@@ -0,0 +1,597 @@
...
+    wormholeFeeCollector,
+    wormholeEmitter,
+    wormholeSequence,
+    authoritySigner,
+    tokenBridgeConfig,
+    custodySigner,
+  };
+};
+
+export const generatePropellerEngineTxns = async (
```
Comment (wormat) [inline relayer/spy_relayer/src/relayer/solana_utils.ts]: This naming sounds like you're using a generator. Suggest:
```suggestion
export const createPropellerEngineTxns = async (
```

File: relayer/spy_relayer/src/relayer/solana_utils.ts [hunk 4]
```diff
@@ -0,0 +1,597 @@
...
+      tokenProgram: splToken.programId,
+      tokenBridge: wormholeAddresses.tokenBridge,
+    })
+    .pubkeys();
+
+  const completeNativeWithPayloadIxs = propellerProgram.methods
+    .propellerCompleteNativeWithPayload()
+    .accounts({
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```
Comment (wormat) [inline relayer/spy_relayer/src/relayer/solana_utils.ts:364]: What's the error here?

**Review by wormat — COMMENTED** [review]
Nice

File: relayer/spy_relayer/src/relayer/solana.ts (hunk 1 above)
Comment (swimricky) [reply relayer/spy_relayer/src/relayer/solana.ts]: this is unnnecessary. he was probably just copying from my engine.test.ts setup

File: relayer/spy_relayer/src/relayer/solana_utils.ts (hunk 4 above)
Comment (swimricky) [reply relayer/spy_relayer/src/relayer/solana_utils.ts:364]: it was some weird thing related to the anchor idl typedefs

Comment (wormat) [reply relayer/spy_relayer/src/relayer/solana_utils.ts:364]: Can you be more specific?

File: relayer/spy_relayer/src/relayer/solana.ts (hunk 1 above)
Comment (swimalick) [reply relayer/spy_relayer/src/relayer/solana.ts]: good catch, removed

File: relayer/spy_relayer/src/relayer/solana_utils.ts (hunk 4 above)
Comment (swimricky) [reply relayer/spy_relayer/src/relayer/solana_utils.ts:364]: this is from a different line but the error should be similar
```
Type 'Partial  >' is not assignable to type 'Accounts{ name: "propeller"; isMut: false; isSigner: false; pda: { seeds: [{ kind: "const"; type: "string"; value: "propeller"; }, { kind: "account"; type: "publicKey"; account: "Propeller"; path: "propeller.swim_usd_mint"; }]; }; } | ... 18 more ... | { ...; }>'.
```

**Comment by swimalick** [comment]
merging for the xhack workshop

**Pull request merged**